docs/design.md

```diff
@@ -138,7 +138,47 @@ upgrades, but must be configured at build time to choose one of these two
 strategies.
 
 In addition to the two image slots, the boot loader requires a scratch area to
-allow for reliable image swapping.
+allow for reliable image swapping. The scratch area must have a size that is
+enough to store at least the largest sector that is going to be swapped. Many
+devices have small equally sized flash sectors, eg 4K, while others have variable
+sized sectors where the largest sectors might be 128K or 256K, so the scratch
+must be big enough to store that. The scratch is only ever used when swapping
+firmware, which means only when doing an upgrade. Given that, there are two main
+reasons for using a larger size for the scratch. One is that the speed of an
```

Comment:
Is this true, though?  Assuming an image that is decently larger than the swap, the same amount of flash is going to be erased and written, whether the swap is 1 sector or more than one. Yes there are fewer swap operations with a larger scratch, but those operations take longer, since the speed concerns (write and erase time) are proportional to the size of the area written or erased.

Reply:
You're right, fixed!